src/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -122,6 +124,96 @@ public static void JsonIgnoreAttribute()
             Assert.Equal(1, obj.MyDictionaryWithIgnore["Key"]);
         }
 
+        [Fact]
+        public static void JsonIgnoreAttribute_UnsupportedCollection()
+        {
+            string json = @"{""MyDict"":{""key"":""value""}}";
+            string wrapperJson = @"{""MyClass"":{""MyDict"":{""key"":""value""}}}";
+
+            // Unsupported collections will throw by default.
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<ClassWithUnsupportedDictionary>(json));
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Serialize(new ClassWithUnsupportedDictionary()));
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<WrapperForClassWithUnsupportedDictionary>(wrapperJson));
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Serialize(new WrapperForClassWithUnsupportedDictionary()));
+
+            // When ignored, we can serialize and deserialize without exceptions.
+            ClassWithIgnoredUnsupportedDictionary obj = JsonSerializer.Deserialize<ClassWithIgnoredUnsupportedDictionary>(json);
```

Comment:
Can you add test for serialize as well (and not just deserialize).

Reply:
Serialize tests are just below.